store/src/java/com/zimbra/cs/pop3/Pop3Message.java

```diff
@@ -71,4 +87,17 @@ boolean isDeleted() {
     String getDigest() {
         return digest;
     }
+
+    public String getUid(Boolean useCustomPop3UID) {
+        if (false == useCustomPop3UID || null == uid || uid.isEmpty()) {
+            // Classic UID string
+            return id + "." + digest;
+        } else {
+            // Custom UID stored in the metadata
+            return uid;
+        }
+    }
+    void setUid(String uid) {
+		this.uid = uid;
+	}
```

Comment:
indentation

Reply:
done

Reply:
it has not been fixed yet... check lines 101-103

Reply:
Plz check now.

Reply:
confirmed it has been fixed. Thanks!